sources/directconnect/connection.go

```diff
@@ -0,0 +1,139 @@
+
+	for _, connection := range output.Connections {
+		attributes, err := sources.ToAttributesCase(connection, "tags")
+		if err != nil {
+			return nil, &sdp.QueryError{
+				ErrorType:   sdp.QueryError_OTHER,
+				ErrorString: err.Error(),
+				Scope:       scope,
+			}
+		}
+
+		item := sdp.Item{
+			Type:            "directconnect-connection",
+			UniqueAttribute: "connectionId",
+			Attributes:      attributes,
+			Scope:           scope,
+			Tags:            tagsToMap(connection.Tags),
+		}
+
+		if connection.LagId != nil {
+			// +overmind:link directconnect-lag
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-lag",
+					Method: sdp.QueryMethod_GET,
+					Query:  *connection.LagId,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// Changes to the lag will affect this
+					In: true,
+					// We can't affect the lag
+					Out: false,
+				},
+			})
+		}
+
+		if connection.Location != nil {
+			// +overmind:link directconnect-location
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-location",
+					Method: sdp.QueryMethod_GET,
+					Query:  *connection.Location,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// Changes to the location will affect this, i.e., its speed, provider, etc.
+					In: true,
+					// We can't affect the location
+					Out: false,
+				},
+			})
+		}
+
+		if connection.LoaIssueTime != nil {
+			// +overmind:link directconnect-loa
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-loa",
+					Method: sdp.QueryMethod_GET,
+					Query:  *connection.ConnectionId,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// Changes to the loa will affect this
+					In: true,
+					// We can't affect the loa
+					Out: false,
+				},
+			})
+		}
+
+		// Virtual Interfaces
+		item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+			// +overmind:link directconnect-virtual-interface
+			Query: &sdp.Query{
+				Type:   "directconnect-virtual-interface",
+				Method: sdp.QueryMethod_LIST,
+				Query:  *connection.ConnectionId,
+				Scope:  scope,
+			},
+			BlastPropagation: &sdp.BlastPropagation{
+				// Changes to the virtual interface won't affect this
+				In: false,
+				// We cannot delete a connection if it has virtual interfaces
+				Out: true,
+			},
+		})
+
+		items = append(items, &item)
+	}
+
+	return items, nil
+}
+
+//go:generate docgen ../../docs-data
+// +overmind:type directconnect-connection
+// +overmind:descriptiveType Connection
+// +overmind:get Get a connection by ID
+// +overmind:list List all connections
+// +overmind:search Search connection by ARN
+// +overmind:group AWS
+// +overmind:terraform:queryMap aws_dx_connection.id
```

Comment:
I went by the `id` here as it is provided as an attribute in terraform resource.
What is the pattern here?

Reply:
any id or arn property that is available on the terraform side is fine. As this is pretty easy to change later, don't sweat it too much.